src/components/Personalization/handlers/createProcessPropositions.js

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+export default ({ schemaProcessors, logger }) => {
+  const wrapRenderWithLogging = (render, item) => () => {
+    return Promise.resolve()
+      .then(render)
+      .then(() => {
+        if (logger.enabled) {
+          logger.info(`Action ${item.toString()} executed.`);
+        }
+        return true;
+      })
+      .catch(error => {
+        if (logger.enabled) {
+          const { message, stack } = error;
+          const errorMessage = `Failed to execute action ${item.toString()}. ${message} ${stack}`;
+          logger.error(errorMessage);
+        }
+        return false;
+      });
+  };
+
+  const renderItems = (renderers, meta) =>
+    Promise.all(renderers.map(renderer => renderer())).then(successes => {
+      // as long as at least one renderer succeeds, we want to add the notification
+      // to the display notifications
+      if (!successes.includes(true)) {
+        return undefined;
+      }
+      return meta;
+    });
+
+  const processItem = item => {
+    const processor = schemaProcessors[item.getSchema()];
+    if (!processor) {
+      return {};
+    }
+    return processor(item);
+  };
+
+  const processItems = ({
+    renderers: existingRenderers,
+    returnedPropositions: existingReturnedPropositions,
+    returnedDecisions: existingReturnedDecisions,
+    items,
+    proposition
+  }) => {
+    let renderers = [...existingRenderers];
+    let returnedPropositions = [...existingReturnedPropositions];
+    let returnedDecisions = [...existingReturnedDecisions];
+    let renderedItems = [];
+    let nonRenderedItems = [];
+    let itemRenderers = [];
+    let atLeastOneWithNotification = false;
```

Comment:
so many flags, a decision diagram would be very helpful for CE engineers debugging an issue.